internal/subscriber/retry/retrier_test.go

```diff
@@ -87,3 +104,69 @@ func Test_Interval_Calculator(t *testing.T) {
 		assert.Equal(t, test.want, actualIntervals)
 	}
 }
+
+func TestRetrier_Start(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	ctxWt, cancel := context.WithTimeout(ctx, time.Second*1)
+	defer cancel()
+	mockBrokerStore := mocks.NewMockIBrokerStore(ctrl)
+	mockCache := mocks2.NewMockICache(ctrl)
+	mockCache.EXPECT().Get(ctxWt, gomock.Any()).Return([]byte{'0'}, nil).AnyTimes()
+	dcs := getMockDelayConsumer(ctx, ctrl)
+	producer := getMockProducer(ctx, ctrl)
+	mockBrokerStore.EXPECT().GetConsumer(
+		ctxWt,
+		gomock.Any(),
+	).Return(dcs, nil).AnyTimes()
+	mockBrokerStore.EXPECT().RemoveConsumer(ctxWt, gomock.Any()).AnyTimes()
+	mockBrokerStore.EXPECT().GetProducer(gomock.Any(), gomock.Any()).Return(producer, nil).AnyTimes()
+
+	r := &Retrier{
+		subscriberID:   "subscription-id",
+		subs:           getMockSubscriptionModel(),
+		bs:             mockBrokerStore,
+		ch:             mockCache,
+		backoff:        NewFixedWindowBackoff(),
+		finder:         NewClosestIntervalWithCeil(),
+		handler:        NewPushToPrimaryRetryTopicHandler(mockBrokerStore),
+		delayConsumers: sync.Map{},
+		errChan:        make(chan error),
+	}
+
+	err := r.Start(ctxWt)
+	assert.Nil(t, err)
+	dcTopics := []string{}
+	r.delayConsumers.Range(func(key, value interface{}) bool {
+		dc := value.(*DelayConsumer)
+		dcTopics = append(dcTopics, dc.topic)
+		return true
+	})
+	sort.Strings(dcTopics)
+	assert.Equal(t, dcTopics, []string{"projects/test-project/topics/test-subscription.delay.30.seconds", "projects/test-project/topics/test-subscription.delay.5.seconds"})
```

Comment:
configure min, and max backoff in a way that all delay topics are returned. And then rather than defining them one by one, use an array

Reply:
I didn't understood the comment

Reply:
here you are only checkin for two delay topics, write test case in a way that all delay topics are returned

Reply:
modified test case to check all delay topics